src/NLog/Targets/FileTarget.cs

```diff
@@ -390,16 +390,14 @@ public string ArchiveNumbering
             get => _archiveNumbering ?? "Sequence";
             set
             {
-                if (string.Equals(value, _archiveNumbering))
-                    return;
-
-                _archiveNumbering = string.IsNullOrEmpty(value) ? null : value.Trim();
-                if (_archiveNumbering is null || string.IsNullOrEmpty(_archiveNumbering))
+                var archiveNumbering = value?.Trim() ?? string.Empty;
+                if (string.IsNullOrEmpty(archiveNumbering) || string.Equals(_archiveNumbering, archiveNumbering, StringComparison.OrdinalIgnoreCase))
                     return;
 
+                _archiveNumbering = archiveNumbering;
                 if (_archiveSuffixFormat is null || ReferenceEquals(_archiveSuffixFormat, _legacyDateArchiveSuffixFormat) || ReferenceEquals(_archiveSuffixFormat, _legacySequenceArchiveSuffixFormat))
                 {
-                    ArchiveSuffixFormat = _archiveNumbering.IndexOf("date", StringComparison.OrdinalIgnoreCase) >= 0 ? _legacyDateArchiveSuffixFormat : _legacySequenceArchiveSuffixFormat;
+                    ArchiveSuffixFormat = archiveNumbering.IndexOf("date", StringComparison.OrdinalIgnoreCase) >= 0 ? _legacyDateArchiveSuffixFormat : _legacySequenceArchiveSuffixFormat;
```

Comment:
```suggestion
                    ArchiveSuffixFormat = archiveNumbering.Contains("date", StringComparison.OrdinalIgnoreCase) ? _legacyDateArchiveSuffixFormat : _legacySequenceArchiveSuffixFormat;
```
this can be simplified

Reply:
Recall that Contains with StringComparison is not available on NET35, which is why using IndexOf. But thank you for noticing.